src/dotty/tools/dotc/ast/tpd.scala

```diff
@@ -363,21 +363,21 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
    *  kind for the given element type in `typeArg`. No type arguments or
    *  `length` arguments are given.
    */
-  def newArray(typeArg: Tree, pos: Position)(implicit ctx: Context): Tree = {
-    val elemType = typeArg.tpe
-    val elemClass = elemType.classSymbol
-    def newArr(kind: String) =
-      ref(defn.DottyArraysModule).select(s"new${kind}Array".toTermName).withPos(pos)
-    if (TypeErasure.isUnboundedGeneric(elemType))
-      newArr("Generic").appliedToTypeTrees(typeArg :: Nil)
-    else if (elemClass.isPrimitiveValueClass)
-      newArr(elemClass.name.toString)
-    else {
-      val typeApplied = newArr("Ref").appliedToTypeTrees(typeArg :: Nil)
-      val classOfArg =
-        ref(defn.Predef_classOf).appliedToTypeTrees(typeArg :: Nil)
-      Apply(typeApplied, classOfArg :: Nil).withPos(typeArg.pos)
-    }
+  def newArray(elemTpe: Type, returnTpe: Type, pos: Position, dims: JavaSeqLiteral)(implicit ctx: Context): Tree = {
```

Comment:
Hum, the signature of this method changes, but I see no call site change in this commit. That's suspicious ;)